comictalker/comictalker.py

```diff
@@ -0,0 +1,222 @@
+        try:
+            alt_covers = self.talker.fetch_alternate_cover_urls(issue_id, issue_page_url)
+            return alt_covers
+        except NotImplementedError:
+            logger.warning(f"{self.talker.source_details.name} has not implemented: 'fetch_alternate_cover_urls'")
+            raise TalkerError(
+                self.talker.source_details.name,
+                4,
+                "The source has not implemented: 'fetch_alternate_cover_urls'",
+            )
+
+    # For issueidentifier
+    def fetch_issues_by_volume_issue_num_and_year(
+        self, volume_id_list: list[int], issue_number: str, year: str | int | None
+    ) -> list[ComicIssue]:
+        try:
+            issue_results = self.talker.fetch_issues_by_volume_issue_num_and_year(volume_id_list, issue_number, year)
+            return issue_results
+        except NotImplementedError:
+            logger.warning(
+                f"{self.talker.source_details.name} has not implemented: 'fetch_issues_by_volume_issue_num_and_year'"
+            )
+            raise TalkerError(
+                self.talker.source_details.name,
+                4,
+                "The source has not implemented: 'fetch_issues_by_volume_issue_num_and_year'",
+            )
+
+    def fetch_issue_cover_urls(self, issue_id: int) -> tuple[str | None, str | None]:
+        try:
+            cover_urls = self.talker.fetch_issue_cover_urls(issue_id)
+            return cover_urls
+        except NotImplementedError:
+            logger.warning(f"{self.talker.source_details.name} has not implemented: 'fetch_issue_cover_urls'")
+            raise TalkerError(
+                self.talker.source_details.name,
+                4,
+                "The source has not implemented: 'fetch_issue_cover_urls'",
+            )
+
+    # Master function to get issue cover. Used by coverimagewidget
+    def async_fetch_issue_cover_urls(self, issue_id: int) -> None:
+        try:
+            # TODO: Figure out async
+            image_url, thumb_url = self.fetch_issue_cover_urls(issue_id)
+            ComicTalker.url_fetch_complete(image_url or "", thumb_url)
+            logger.info("Should be downloading image: %s  thumb: %s", image_url, thumb_url)
+            return
+
+            # Should be all that's needed? CV functions will trigger everything.
+            self.talker.async_fetch_issue_cover_urls(issue_id)
+        except NotImplementedError:
+            logger.warning(f"{self.talker.source_details.name} has not implemented: 'async_fetch_issue_cover_urls'")
+
+    # Used by coverimagewidget.start_alt_cover_search
+    def fetch_issue_page_url(self, issue_id: int) -> str | None:
+        try:
+            page_url = self.talker.fetch_issue_page_url(issue_id)
+            return page_url
+        except NotImplementedError:
+            logger.warning(f"{self.talker.source_details.name} has not implemented: 'fetch_issue_page_url'")
+            return None
```

Comment:
We don't need this to be a `ComicTalker` function this can be an internal function for `ComicVineTalker` just call it right before fetch_alternate_cover_urls